File: Coordenada.cc
#include "Coordenada.h"

using namespace std;

Coordenada::Coordenada(){
    grados=-1;
    minutos=-1;
    direccion='X';
    gps=0;
}

Coordenada::Coordenada(int g,int m,char p){
    grados=g;
    minutos=m;
    direccion=p;
    gps=grados+(minutos/60);
    if(direccion=='S'||direccion=='O'){
        gps=gps*-1;
    }
}

Coordenada::Coordenada(const Coordenada &p){
    
    if(this!=&p){
            grados=p.grados;
            minutos=p.minutos;
            direccion=p.direccion;
            gps=p.gps;
    }
}

Coordenada::~Coordenada(){
    grados=-1;
    minutos=-1;
    direccion='X';
    gps=0;
}

Coordenada& Coordenada::operator =(const Coordenada& t){
    if(this!=&t){
            grados=t.grados;
            minutos=t.minutos;
            direccion=t.direccion;
            gps=t.gps;
    }
    return *this;
}

int Coordenada::getGrados(){
    return grados;
}
int Coordenada::getMinutos(){
    return minutos;
}
char Coordenada::getDireccion(){
    return direccion;
}
double Coordenada::getGps(){
    return gps;
}

ostream & operator <<(ostream &os, const Coordenada &c){
    os << c.grados << " " <<c.minutos<<" "<<c.direccion;
    return os;
}

File: Coordenada.java
public class Coordenada{
// CHRISTIAN CAMELO ESPINAL AW388156
    private int grados;
    private int minutos;
    private char direccion;
    private double gps;

    public Coordenada(){
        grados =-1;minutos=-1;direccion='x';gps=-1;
    }

    public Coordenada(int g, int m, char p){
        grados=g;minutos=m;direccion=p;
        gps = grados +(minutos/60.0);
        if(direccion=='S'||direccion=='O')gps=gps*-1;
    }

    public int getGrados(){
        //System.out.println("            Devuelvo grados desde Coor "+grados);
        return grados;}
    public int getMinutos(){return minutos;}
    public char getDireccion(){return direccion;}
    public double getGps(){
        //System.out.println("            Devuelvo Gps desde coord"+gps);
        return gps;}
}